net/websocat/Portfile

```diff
@@ -0,0 +1,716 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+PortGroup           cargo 1.0
+
+github.setup        vi websocat 1.4.0 v
+
+categories          net
+license             MIT
+platforms           darwin
+maintainers         {gmail.com:herby.gillot @herbygillot} openmaintainer
+
+description         Netcat, curl and socat for WebSockets.
+long_description    Command-line client for web sockets, like \
+                    netcat/curl/socat for ws://. 
+
+destroot {
+    xinstall -m 755 ${worksrcpath}/target/[cargo.rust_platform]/release/${name} ${destroot}${prefix}/bin/
+    xinstall -d 755 ${destroot}${prefix}/share/doc/${name}
+    xinstall -m 644 ${worksrcpath}/doc.md ${destroot}${prefix}/share/doc/${name}/
+}
+
+cargo.crates \
+	MacTypes-sys		2.1.0 eaf9f0d0b1cc33a4d2aee14fb4b2eac03462ef4db29c8ac4057327d8a71ad86f \
+	arc-swap		0.3.7 1025aeae2b664ca0ea726a89d574fe8f4e77dd712d443236ad1de00379450cf6 \
+	arrayvec		0.4.10 92c7fb76bc8826a8b33b4ee5bb07a247a81e76764ab4d55e8f73e3a4d8808c71 \
```

Comment:
I know i have made this comment before for cargo based ports, but still, having to hard code all these versions and hashcodes seems destined to make this port tough to maintain/updated...

Reply:
I 10000000000% agree 😐

Reply:
Additionally, the `cargo` portgroup needs to be tweaked:

Crates are currently specified with a *sha256* hash which are then used to auto-generate checksums for each crate file. However Macports now requires *sha256* and *rmd160*, meaning the checksums created by the cargo portgroup are invalid.

As a result, I manually specify all the checksums in the expected format for both the main distfile and all crates.

Reply:
So... if you just removed the rmd160 sums from the portfile, you wouldn’t need to hardcode all the sha256 ones ?

Reply:
> So... if you just removed the rmd160 sums from the portfile, you wouldn’t need to hardcode all the sha256 ones ?

Yes, correct, except for the checksum block needed for the actual main websocat distfile itself.

I just made a change where I use `checksum-append` to only add the `rmd160` and `size` checksums to the `sha256` sums already being specified in `cargo.crates`.  Seemed to work locally, let's see if it passes CI.  With this change, at least now we are only specifying the `sha256` sum once.

Reply:
Thanks. What I was getting at was if removing the rmd160 sums completely works (I am not sure if base demands these are there ?) then I would suggest Maybe in this case that would be acceptable.

How do you make the missing checksums by hand for each, or via some script ?

Reply:
When I wrote the first version of this portfile, I did not have the second checksum section that specified `rmd160` and `size`.  The Portfile only had the `cargo.crates` section, so in effect, only SHA256 sums were available.  This cause `port` to complain and print out a suggested `checksums` block that had the full `sha256`, `rmd160` and `size` for all distfiles, both the main tarball + all the crates.

I wrote a quite Python script to verify that the *suggested* checksum blocks had the same SHA256 sum as the blocks already present in the `cargo.crates` section, which are sourced from a `Cargo.lock` file found in Cargo projects.

It was only actually reading the `cargo` portgroup source that I discovered that the `cargo_fetch` portgroup suggests a Perl one-liner as well as an available utility to help in generating these:

https://github.com/macports/macports-ports/blob/07361deae5c1eb4a8b08fdc77c59d2a1e7dd477b/_resources/port1.0/group/cargo_fetch-1.0.tcl#L17-L29

At the moment, this Portfile is optimal as it is now since it specifies all three recommended checksum types for all distfiles, for the main tarball as well as all required crates.